### DISCUSSION TITLE
GSoC intro

### DISCUSSION THREAD

**Discussion by bsh90** [networkx/networkx#4696] [Ideas]
Dear Mridul Seth and Dan Schult,
I am interested in implementing one of algorithms or improving test-cases related to network science for GSoC of this summer(Such as Community Detection Algorithms). I am a student of master physics of FU university of berlin. My student day job is in software testing. i worked with Java and Python. I am interested in network science and system science and would be happy to learn more about it in this project.

**Comment by MridulS** [comment]
Hi @bsh90!

As a first step you could go through the ideas list http://networkx.org/documentation/latest/developer/projects.html#mentored-projects and see if you are interested in any of them. (You can also propose your own idea for a NetworkX project!)

To get acquainted with our development process please go through https://networkx.org/documentation/latest/developer/contribute.html#dev-workflow to setup
your local dev environment and https://networkx.org/documentation/latest/developer/new_contributor_faq.html for
some common FAQs.

For some issues to work on to get familiarised with NetworkX you can look at
https://github.com/networkx/networkx/issues?q=is%3Aissue+is%3Aopen+label%3ASprint or you can pick any
other issue on the GitHub issue tracker https://github.com/networkx/networkx/issues.

If you interested in the notebooks project you can also look at networkx/notebooks#13

**Reply by bsh90** [reply]
Dear @MridulS 
Edit for documentation:
For pipenv environment option, i am using the lines bellow:
```
pipenv install -r requirements/default.txt
pipenv install -r requirements/developer.txt
pipenv install -r requirements/test.txt
```

**Reply by Gopal-Dahale** [reply]
@bsh90 Are you working in windows?

**Reply by bsh90** [reply]
@Gopal-Dahale  No, I am on Linux, debian.
